tests/Agent/IntegrationTests/IntegrationTestHelpers/AgentLogFile.cs

```diff
@@ -49,10 +49,12 @@ public AgentLogFile(string logDirectoryPath, ITestOutputHelper testLogger, strin
                 }
 
                 Thread.Sleep(TimeSpan.FromSeconds(1));
-            } while (timeTaken.Elapsed < timeout);
+            } while (timeTaken.Elapsed < timeout && logFileExpected);
```

Comment:
I wonder if you couldn't wrap the entire `do...while` and `throw` with `if (logfileExpected)`? Might be a tad cleaner.